src/export/InstanceExporter.ts

```diff
@@ -175,13 +181,39 @@ export class InstanceExporter implements Fishable {
       }
     });
 
-    const paths = ['', ...rules.map(rule => rule.path)];
+    const paths = [
+      '',
+      ...rules.map(rule =>
+        assembleFSHPath(ruleMap.get(rule.path)?.pathParts ?? []).replace(/\[0\]/g, '')
+      )
+    ];
     // To correctly assign properties, we need to:
-    // 1 - Assign implied properties on the original instance
-    // 2 - Assign rule properties on a copy of the result of 1, so that rule assignment can build on implied assignment
-    // 3 - Merge the result of 2 with the result of 1, so that any implied properties which may have been overwrtiten
-    //     in step 2 are maintained...don't worry I'm confused too
-    setImpliedPropertiesOnInstance(instanceDef, instanceOfStructureDefinition, paths, this.fisher);
+    // 1 - Create useful slices for rules so that properties are assigned in the correct places in arrays
+    // 2 - Assign implied properties on the original instance
+    // 3 - Assign rule properties on a copy of the result of 2, so that rule assignment can build on implied assignment
+    // 4 - Merge the result of 3 with the result of 2, so that any implied properties which may have been overwritten
+    //     in step 3 are maintained...don't worry I'm confused too
```

Comment:
So technically, we did not introduce the confusing things, they've always been confusing! 😆 We just added in a new step at step 1. But we did revisit this comment just now, and @mint-thompson and I have decided that we can actually safely eliminate step 4. I believe this used to be necessary to handle complex assignments properly, but @mint-thompson (separately from this PR) recently(ish) improved how we assign complex values and since that handles assignments properly now, we no longer need to use the copy of the instance or merge back the result. So we will updated the comment and removed step 4 from the code.

Reply:
See [f8f39d8](https://github.com/FHIR/sushi/pull/1121/commits/f8f39d82d6da71f6b559615da78829f82ed79245).